src/lib/rules/html-checker/html-checker.ts

```diff
@@ -0,0 +1,107 @@
+
+const debug = d(__filename);
+
+// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        /** The promise that represents the scan by html checker. */
+        let htmlCheckerPromise: Promise<any>;
+        /** The options to pass to the html checker. */
+        const scanOptions = {
+            data: '',
+            format: 'json'
+        };
+
+        type Error = { // eslint-disable-line no-unused-vars
+            extract: string, // code snippet
+            firstColumn: number,
+            lastLine: number,
+            hiliteStart: number,
+            message: string,
+            subType: string
+        };
+
+        const start = (data: ITargetFetchEnd) => {
+            const { response } = data;
+            const htmlChecker = require('html-validator');
+
+            scanOptions.data = response.body.content;
+            htmlCheckerPromise = htmlChecker(scanOptions);
+        };
+
+        const end = async (data: IScanEnd) => {
+            const { resource } = data;
+            let result;
+
+            if (!htmlCheckerPromise) {
+                return;
+            }
+
+            debug(`Waiting for Html Checker results for ${resource}`);
+
+            try {
+                result = await htmlCheckerPromise;
+            } catch (e) {
+                debug(`Error getting html checker result for ${resource}.`);
+                await context.report(resource, null, `Couldn't get results from Html Checker for ${resource}.`);
+
+                return;
+            }
+
+            debug(`Received Html Checker results for ${resource}`);
+
+            const reportPromises = result.messages.map((error: Error): Promise<void> => {
+                const position: IProblemLocation = {
+                    column: error.firstColumn,
+                    elementColumn: error.hiliteStart + 1,
+                    elementLine: 1, // We will pass in the single-line code snippet generated from the html checker, so the elementLine is always 1
+                    line: error.lastLine
+                };
+
+                return context.report(resource, null, error.message, null, position, Severity[error.subType], error.extract);
+            });
+
+            try {
+                await Promise.all(reportPromises);
+            } catch (e) {
+                debug(`Error reporting the html checker results.`);
+
+                return;
+            }
+        };
+
+        return {
+            'scan::end': end,
+            'targetfetch::end': start
+        };
+    },
+    meta: {
+        docs: {
+            category: 'Interoperability',
+            description: 'Validating html using `the Nu html checker`'
+        },
+        fixable: 'code',
+        recommended: true,
+        schema: [],
```

Comment:
Doesn't the checker allow any configuration? I thought we could ignore some type of errors, etc.